medallion/filters/basic_filter.py

```diff
@@ -155,93 +151,64 @@ def filter_by_version(data, version):
-                        add = False
-                if add:
-                    match_objects.append(obj)
-        return match_objects
+                    s2 = determine_spec_version(match)
+                if obj["id"] == match["id"] and s2 > s1:
+                    add = False
+            if add:
+                return True
+        return False
 
     def process_filter(self, data, allowed=(), manifest_info=(), limit=None):
-        filtered_by_type = []
-        filtered_by_id = []
-        filtered_by_spec_version = []
-        filtered_by_added_after = []
         filtered_by_version = []
         final_match = []
         save_next = []
         headers = {}
-
+        match_objects = []
         # match for type and id filters first
-        match_type = self.filter_args.get("match[type]")
-        if match_type and "type" in allowed:
-            filtered_by_type = self.filter_by_type(data, match_type)
-        else:
-            filtered_by_type = copy.deepcopy(data)
-
-        match_id = self.filter_args.get("match[id]")
-        if match_id and "id" in allowed:
-            filtered_by_id = self.filter_by_id(filtered_by_type, match_id)
-        else:
-            filtered_by_id = filtered_by_type
-
-        # match for added_after
-        added_after_date = self.filter_args.get("added_after")
-        if added_after_date:
-            filtered_by_added_after = self.filter_by_added_after(filtered_by_id, manifest_info, added_after_date)
-        else:
-            filtered_by_added_after = filtered_by_id
-
-        # match for spec_version
-        match_spec_version = self.filter_args.get("match[spec_version]")
-        if "spec_version" in allowed:
-            filtered_by_spec_version = self.filter_by_spec_version(filtered_by_added_after, match_spec_version)
+        if (self.match_type and "type" in allowed) or (self.match_id and "id" in allowed) \
+           or (self.added_after_date) or ("spec_version" in allowed):
+            for obj in data:
+                if self.match_type and "type" in allowed:
+                    if not (any(s == obj.get("type") for s in self.type_)) and not (any(s == obj.get("id").split("--")[0] for s in self.type_)):
+                        continue
+                if self.match_id and "id" in allowed:
+                    if not ("id" in obj and any(s == obj["id"] for s in self.id_)):
+                        continue
+
+                if self.added_after_date:
+                    if not self.check_added_after(obj, manifest_info, self.added_after_date):
+                        continue
+
+                if "spec_version" in allowed:
+                    if not self.check_by_spec_version(obj, self.match_spec_version, data):
+                        continue
+                match_objects.append(obj)
         else:
-            filtered_by_spec_version = filtered_by_added_after
-
+            match_objects = copy.deepcopy(data)
```

Comment:
Do we need to copy here? I know the previous code did, but now that we're deepcopying in `get_object` and `get_objects` might be able to just do this here:
```suggestion
            match_objects = data
```